workshops/reddit_viewer/README.md

```diff
@@ -0,0 +1,126 @@
+
+# Step 4: Getting and handling user input
+
+Go to the index.html file and add a `form` with a `method="POST"`. In the form area add an `<h3>` that just has something like `Go to subreddit`. In the same form area add an input with the `name="subreddit"`. Also have the `input type=submit`. Thats it from the html side of user input. Go to the main.py file and add another @app.route going to the ('/') but this time add a comma and have a methods=["POST."] `@app.route('/', methods=['POST'])`. This means that when the html sends a POST event, we will handle it in the function. Under the app.route, add a new method called `front_page_post`. In that function remember to put a `return render_template('index.html')` put keep this at the bottom of the function always.
+
+Above the return statement add a new variable called subreddit and make it equal to `request.form['subreddit'] `. This is basically just capturing the user input from the text box we made.
+
+![The second app.route method](https://cloud-r6o8ztkzw.vercel.app/image.png).
+
+# Step 5: Showing the posts
+
+Now we want to have the html output whatever posts the subreddit has. So, go into the index.html file and add these lines. `{% block content %} {% endblock %}`. These are called Jinja Templates and it's what Flask uses to dynamically change the HTML. Block content just means that each part is like a block and we want it to be shown like that.
+
+In between your block content sections add this line, `{% for submission in submissions %}`. This just means that every post in the subreddit should have the same html applied to it. The html that's applied to it whatever is in between it.
+
+Currently we do not have a "submissions" variable. That is easy to create. Go to the main.py file and in your return statement in the \_post method add a comma after the index.html and add this line. `submissions=reddit.subreddit(subreddit.upper()).hot(limit=100)`. This is just saying that the submissions variable in the html should be all the posts from the user defined subreddit (which is regardless of case because of the .upper). Then we sort them by hot, and only showing the top 100 posts.
+
+That is all we need for the python part. Now we just need to finish off the html. In your for-statement section in the html file, make an `<a>` element with an `href={{ "https://reddit.com" + submission['permalink']}}`. This just means that we want the title to go to the actual reddit post with comments when we click on it. We must set it with reddit.com or else it will try to use whatever website we are currently on.
+
+Now in the `<a>` section add an `<h2>` that in between it is `{{ submission['title']. This just means that we can see the posts title.
+
+Now under the `<a>` add an img tag and make its `src={{ submission['url']`. This is because the URL of the post we are getting is also the image.
+
+Now run your program and you should be able to enter a subreddit name and see its posts.
```

Comment:
Same for all of step 5

Reply:
I just removed step 5 because it was too confusing lol!